### PR TITLE
feature/edit match validation

### LINKED ISSUE #91: Allow modifying of the match verification status
Introduce `/api/matches/verification-status PATCH` to accept changing of the match's `MatchVerificationStatus` value.

### PR DESCRIPTION
**Pull request by bugg86**
Closes #91

Adds a new PATCH endpoint for editing the match verification status of a match.

## Other Changes
- A few miscellaneous code warning cleanups.
- Added helper methods from MatchService -> MatchRepository to edit just the match verification status value.

### CODE REVIEW

File: API/Repositories/Interfaces/IMatchesRepository.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public interface IMatchesRepository : IRepository<Match>
 	Task<Match?> GetFirstMatchNeedingAutoCheckAsync();
 	Task<IList<Match>> GetNeedApiProcessingAsync();
 	Task<IEnumerable<Match>> GetByMatchIdsAsync(IEnumerable<long> matchIds);
+	Task<Match> UpdateVerificationStatus(int id, int? verificationStatus);
```
Comment (hburn7) [inline API/Repositories/Interfaces/IMatchesRepository.cs:18]: `verificationStatus` should be an `int` on update (not `int?`). I'm honestly not sure if there is ever a case where `verificationStatus` is null, will have to investigate separately.

**Review by hburn7 — CHANGES_REQUESTED** [review]

File: API/Repositories/Interfaces/IMatchesRepository.cs (hunk 1 above)
Comment (bugg86) [reply API/Repositories/Interfaces/IMatchesRepository.cs:18]: Gotcha, I think changing the method to accept an int and using the null forgiving operator when passing it in should probably be fine for now. Like this: `(int)match.VerificationStatus!`.

Comment (hburn7) [reply API/Repositories/Interfaces/IMatchesRepository.cs:18]: Null ignore should be fine but doesn't feel optimal.

Comment (hburn7) [reply API/Repositories/Interfaces/IMatchesRepository.cs:18]: I'll check db and see if this field should be nullable.

**Comment by hburn7** [comment]
Confirmed that `matchVerificationStatus` can be made non-null. There are no null values in the database.

**Comment by bugg86** [comment]
> Confirmed that `matchVerificationStatus` can be made non-null. There are no null values in the database.

Alright do you want me to change the entity to be non-null in this PR?

**Review by hburn7 — APPROVED** [review]

**Comment by hburn7** [comment]
> > Confirmed that `matchVerificationStatus` can be made non-null. There are no null values in the database.
> 
> Alright do you want me to change the entity to be non-null in this PR?

Considering everything is working correctly, let's address in a separate PR.

**Pull request merged**